### PR TITLE
feat(api) new endpoint: FindConnectors

### PR DESCRIPTION
**Pull request by a-launois**
## Description

New API configuration endpoint:
```
GET /configuration/connectors
```

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x (master)
- [x] 24.04.x

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Connector/Application/UseCase/FindConnectors/FindConnectors.php [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+            $connectors = $this->readConnectorRepository->findByRequestParametersAndCommandTypes(
+                $this->requestParameters,
+                $this->retrieveConnectorTypesBasedOnContactRights(),
+            );
+
+            $commandIds = [];
+            foreach ($connectors as $connector)
+            {
+                array_push($commandIds, ...$connector->getcommandIds());
```
Comment (abuathier) [inline centreon/src/Core/Connector/Application/UseCase/FindConnectors/FindConnectors.php:81]: Method case is not required in php but recommended.
```suggestion
                array_push($commandIds, ...$connector->getCommandIds());
```

File: centreon/src/Core/Connector/Application/UseCase/FindConnectors/FindConnectors.php [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            $response->connectors[] = $connectorDto;
+        }
+
+        return $response;
+    }
+
+    /**
+     * @return CommandType[]
+     */
+    private function retrieveConnectorTypesBasedOnContactRights(): array
```
Comment (abuathier) [inline centreon/src/Core/Connector/Application/UseCase/FindConnectors/FindConnectors.php:133]: This method is 40 LoC and is 100% duplicate from the `FindCommands` use case.
Maybe we can do better like cerating a Domain Right class or something like that ?

File: centreon/src/Core/Connector/Infrastructure/API/FindConnectors/FindConnectorsPresenter.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            $this->setResponseStatus($response);
+        } else {
+            $result = [];
+
+            foreach ($response->connectors as $connector) {
+                $result[] = [
+                    'id' => $connector->id,
+                    'name' => $connector->name,
+                    'command_line' => $connector->commandLine,
+                    'description' => self::emptyStringAsNull($connector->description),
```
Comment (abuathier) [inline centreon/src/Core/Connector/Infrastructure/API/FindConnectors/FindConnectorsPresenter.php:58]: It works but this is not correct to call statically an instance method
```suggestion
                    'description' => $this->emptyStringAsNull($connector->description),
```

File: centreon/src/Core/Connector/Infrastructure/Repository/DbReadConnectorRepository.php [hunk 4]
```diff
@@ -55,4 +72,87 @@ public function exists(int $id): bool
...
+            );
+            $sqlConcatenator->storeBindValueMultiple(
+                ':command_type',
+                array_map(
+                    fn(CommandType $commandType): int => CommandTypeConverter::toInt($commandType),
+                    $commandTypes
+                ),
+                \PDO::PARAM_INT
+            );
+        }
```
Comment (abuathier) [inline centreon/src/Core/Connector/Infrastructure/Repository/DbReadConnectorRepository.php:133]: Same as previous
```suggestion
        $sqlConcatenator->appendGroupBy(
            <<<'SQL'
                    GROUP BY `connector`.id
                    SQL
            );
        $sqlConcatenator->storeBindValueMultiple(
            ':command_type',
            array_map(
                fn(CommandType $commandType): int => CommandTypeConverter::toInt($commandType),
                $commandTypes
            ),
            \PDO::PARAM_INT
        );
```

File: centreon/src/Core/Connector/Domain/Model/Connector.php [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        private string $name,
+        private string $commandLine,
+        private string $description = '',
+        private array $commandIds = [],
+        private bool $isActivated = true,
+    )
+    {
+        Assertion::positiveInt($id, 'Connector::id');
+
+        Assertion::notEmptyString($name, 'Connector::name');
```
Comment (callapa) [inline centreon/src/Core/Connector/Domain/Model/Connector.php:58]: I think you've forgotten about the "trim" function before

File: centreon/src/Core/Connector/Domain/Model/Connector.php [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        Assertion::maxLength($name, self::NAME_MAX_LENGTH, 'Connector::name');
+
+        Assertion::notEmptyString($commandLine, 'Connector::commandLine');
```
Comment (callapa) [inline centreon/src/Core/Connector/Domain/Model/Connector.php:61]: I think you've forgotten about the "trim" function before

File: centreon/src/Core/Connector/Infrastructure/Repository/DbReadConnectorRepository.php [hunk 7]
```diff
@@ -55,4 +72,87 @@ public function exists(int $id): bool
...
+     */
+    public function findByRequestParametersAndCommandTypes(
+        RequestParametersInterface $requestParameters,
+        array $commandTypes
+    ): array
+    {
+        if ([] === $commandTypes) {
+            return [];
+        }
+
```
Comment (a-launois) [inline centreon/src/Core/Connector/Infrastructure/Repository/DbReadConnectorRepository.php:87]:
```suggestion
```

File: centreon/src/Core/Connector/Infrastructure/Repository/DbReadConnectorRepository.php [hunk 8]
```diff
@@ -55,4 +72,87 @@ public function exists(int $id): bool
...
+            'name' => 'connector.name',
+            'command.id' => 'command.command_id',
+            'command.name' => 'command.command_name',
+        ]);
+
+        if ($commandTypes !== []) {
+            $commandTypeFilter = <<<'SQL'
+                AND `command`.command_type IN (:command_type)
+                SQL;
+        }
```
[the remark this answers is not in the document]
Comment (a-launois) [reply centreon/src/Core/Connector/Infrastructure/Repository/DbReadConnectorRepository.php:101]: Removed the line 84

**Review by arnapou — DISMISSED** [review]

**Review by callapa — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**